lib/msal-browser/src/utils/crypto/BrowserCrypto.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { ICrypto, PkceCodes } from "msal-common";
+import { GuidGenerator } from "../../math/GuidGenerator";
+import { Base64Encode } from "../../math/Base64Encode";
+import { Base64Decode } from "../../math/Base64Decode";
+import { PkceGenerator } from "../../math/PkceGenerator";
+
+export class BrowserCrypto implements ICrypto {
+
+    private guidGenerator: GuidGenerator;
+    private b64Encode: Base64Encode;
+    private b64Decode: Base64Decode;
+    private pkceGenerator: PkceGenerator;
+
+    constructor() {
+        this.b64Encode = new Base64Encode();
+        this.b64Decode = new Base64Decode();
+        this.guidGenerator = new GuidGenerator(BrowserCrypto.getBrowserCryptoObj());
+        this.pkceGenerator = new PkceGenerator(BrowserCrypto.getBrowserCryptoObj());
+    }
+
+    /**
+     * Creates a new random GUID - used to populate state?
+     * @returns string (GUID)
+     */
+    createNewGuid(): string {
+        return this.guidGenerator.generateGuid();
+    }
+
+    base64Encode(input: string): string {
+        return this.b64Encode.encode(input);
+    }    
+    
+    base64Decode(input: string): string {
+        return this.b64Decode.decode(input);
+    }
+
+    async generatePkceCodes(): Promise<PkceCodes> {
+        return this.pkceGenerator.generateCodes();
+    }
+
+    static getBrowserCryptoObj(): Crypto {
+        return "msCrypto" in window ? window["msCrypto"] : window.crypto;
```

Comment:
FYI This works for `getRandomValues`, but does not work for the other Crypto APIs. I personally think the better solution is to create abstractions for each Crypto API, then just an abstraction for the entire crypto interface.

Reply:
I would almost rather use the webcrypto shim than have diverging code for msCrypto and crypto. It's frustrating to get these working side by side, and we only use crypto for `getRandomValues` and `crypto.subtle.digest`.

Reply:
Resolved